src/core/middleware.js

```diff
@@ -45,6 +50,19 @@ export function prefixMiddleWare(req, res, next, { _config = config } = {}) {
     // We skip prepending an app if we'd previously prepended a lang and the
     // 2nd part of the URL is now a valid app.
     log.info('Application in URL is valid following prepending a lang.');
+  } else if (
+    hasValidUrlException || isValidUrlException(URLParts[1], { _config })
```

Comment:
I think it's a side effect of checking for `isValidUrlException(URLParts[1], { _config })` but I'll see about adding a test that fails if it's gone. It could be that it's unneeded because at this point `URLParts[1]` should always be the app/exceptionURL so you're right it might be erroneous. If it is I'll change around the code so it's easier to follow; like you mentioned this middleware is a bit hairy to grok.